iml-agent/src/action_plugins/check_ha.rs

```diff
@@ -2,94 +2,234 @@
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
 
-use crate::agent_error::ImlAgentError;
+use crate::{agent_error::ImlAgentError, systemd};
 use elementtree::Element;
-use futures::{future, Future, TryFutureExt};
+use futures::try_join;
 use iml_cmd::cmd_output_success;
-use std::{collections::HashMap, fmt};
-
-/// standard:provider:ocftype (e.g. ocf:heartbeat:ZFS, or stonith:fence_ipmilan)
-#[derive(serde::Deserialize, serde::Serialize, Clone, Debug)]
-pub struct ResourceAgentType {
-    pub standard: String,         // e.g. ocf, lsb, stonith, etc..
-    pub provider: Option<String>, // e.g. heartbeat, lustre, chroma
-    pub ocftype: String,          // e.g. Lustre, ZFS
+use iml_wire_types::{
+    ComponentState, ConfigState, ResourceAgentInfo, ResourceAgentType, ServiceState,
+};
+use std::collections::HashMap;
+use tokio::fs::metadata;
+
+fn create(elem: &Element, group: Option<String>) -> ResourceAgentInfo {
```

Comment:
This could be:

`fn create<'a>(elem: &Element, group: impl Into<Option<&'a str>>) -> ResourceAgentInfo`

which would simplify callers.

Reply:
Or it's okay as it is...  The callers are not that complicated.

Reply:
Yeah, it works fine without it. I think it makes more sense in the big picture to push the complexity into trait bounds as the code either needs to repeat the same patterns each time or have them handled by the bound.